python/src/vmaf/tools/ResultAssembler.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+
+from collections import OrderedDict
+from xml.etree import ElementTree
+import json
+import copy, numpy as np
+
+__copyright__ = "Copyright 2016-2018, Netflix, Inc."
+__license__ = "Apache, Version 2.0"
+
+
+def print_usage():
+    print("usage: python " + os.path.basename(sys.argv[0]) + " input_file output_file\n")
+    print("input_file contains a list of files for assembly (can be xml or json)")
+    print("output_file contains the assembled file")
+
+class FileAssembler():
+
+    SUPPORTED_FILE_TYPES = ['xml', 'json']
+
+    def __init__(self, to_assemble_input):
+        self.aggregate_method = ''
+        self.executorId = ''
+        self.assembled_frame_dict_list = []
+        self.scores_dict = OrderedDict()
+        self.to_assemble_input = to_assemble_input
+        self.to_assemble_list = []
+
+    def _parse_files(self):
+        pass
+
+    def _write_assembled(self):
+        pass
+
+    def create_assembly_filelist(self):
+
+        if isinstance(self.to_assemble_input, list):
+            self.to_assemble_list = self.to_assemble_input
+        else:
+            with open(self.to_assemble_input, "rt") as input_file:
+                for line in input_file.readlines():
+
+                    # match comment
+                    mo = re.match(r"^#", line)
+                    if mo:
+                        print("Skip commented line: {}".format(line))
+                        continue
+
+                    # match whitespace
+                    mo = re.match(r"[\s]+", line)
+                    if mo:
+                        continue
+
+                    mo = re.match(r"([\S]+)", line)
+                    if not mo:
+                        print("Invalid file: {}".format(line))
+                        print_usage()
+                        return 1
+
+                    self.to_assemble_list.append(line.strip())
+
+    def assemble(self):
+        """
+        Main file assembly logic
+        """
+        self.create_assembly_filelist()
+        self._assert()
+        self._parse_files()
+        assembly_output = self._write_assembled()
+
+        return assembly_output
+
+    def create_aggregation_dict(self):
+
+        # TODO fix: only does for aggregation
+        aggregate_dict = OrderedDict()
+        valid_score_keys = sorted([key for key in self.scores_dict if key.endswith("score")])
+        for key in valid_score_keys:
+            aggregate_dict[key] = np.mean(self.scores_dict[key])
```

Comment:
Here you hard-coded the aggregate method to 'mean'. You shouldn't. See how Result.to_xml() does it at:
https://github.com/Netflix/vmaf/blob/master/python/src/vmaf/core/result.py#L221